packages/devtools_app/lib/src/screens/memory/panes/leaks/controller.dart

```diff
@@ -6,78 +6,81 @@ import 'dart:async';
-const _extensionKindToReceiveLeaksDetails = 'memory_leaks_details';
+import 'primitives/simple_items.dart';
 
 const yamlFilePrefix = 'memory_leaks';
 
 class LeaksPaneController {
-  LeaksPaneController() {
-    _subscribeForMemoryLeaksMessages();
+  LeaksPaneController()
+      : assert(
+          supportedLeakTrackingProtocols
+              .contains(appLeakTrackerProtocolVersion),
+        ) {
+    subscriptionWithHistory = serviceManager
+        .service!.onExtensionEventWithHistory
+        .listen(_onAppMessageWithHistory);
   }
 
-  final status = AnalysisStatusController();
+  final analysisStatus = AnalysisStatusController();
 
   final leakSummaryHistory = ValueNotifier<String>('');
-  final leakSummaryReceived = ValueNotifier<bool>(false);
+  late String appProtocolVersion;
+  final appStatus =
+      ValueNotifier<AppStatus>(AppStatus.noCommunicationsRecieved);
+
   LeakSummary? _lastLeakSummary;
 
   final _exportController = ExportController();
 
-  late StreamSubscription summarySubscription;
-  late StreamSubscription detailsSubscription;
-
-  /// Subscribes for summary with history and for details without history.
-  void _subscribeForMemoryLeaksMessages() {
-    detailsSubscription =
-        serviceManager.service!.onExtensionEvent.listen(_receivedLeaksDetails);
-
-    summarySubscription = serviceManager.service!.onExtensionEventWithHistory
-        .listen(_receivedLeaksSummary);
-  }
+  late StreamSubscription subscriptionWithHistory;
 
   void dispose() {
-    unawaited(summarySubscription.cancel());
-    unawaited(detailsSubscription.cancel());
-    status.dispose();
+    unawaited(subscriptionWithHistory.cancel());
+    analysisStatus.dispose();
   }
 
-  void _receivedLeaksSummary(Event event) {
-    if (event.extensionKind != _extensionKindToReceiveLeaksSummary) return;
-    leakSummaryReceived.value = true;
-    try {
-      final newSummary = LeakSummary.fromJson(event.json!['extensionData']!);
-      final time = event.timestamp != null
-          ? DateTime.fromMicrosecondsSinceEpoch(event.timestamp!)
-          : DateTime.now();
+  static DateTime _fromEpochOrNull(int? microsecondsSinceEpoch) =>
+      microsecondsSinceEpoch == null
+          ? DateTime.now()
+          : DateTime.fromMicrosecondsSinceEpoch(microsecondsSinceEpoch);
+
+  void _onAppMessageWithHistory(Event vmServiceEvent) {
+    if (appStatus.value == AppStatus.unsupportedProtocolVersion) return;
+
+    final message = EventFromApp.fromVmServiceEvent(vmServiceEvent)?.message;
+    if (message == null) return;
+
+    if (message is LeakTrackingStarted) {
+      appStatus.value = AppStatus.leakTrackingStarted;
+      appProtocolVersion = message.protocolVersion;
+      return;
+    }
+
+    if (message is LeakSummary) {
+      appStatus.value = AppStatus.leaksFound;
+      if (message.matches(_lastLeakSummary)) return;
+      _lastLeakSummary = message;
+
+      final time = _fromEpochOrNull(vmServiceEvent.timestamp).toLocal();
```

Comment:
Have you actually encountered a `null` timestamp? The [service spec](https://github.com/dart-lang/sdk/blob/master/runtime/vm/service/service.md#event) states that this field is always provided for `Event`.